### PR TITLE
add backlog message items field to Slack Plugin

### PR DESCRIPTION
**Pull request by lingpri**
slack_alert_docs

### CODE REVIEW

**Comment by jalogisch** [comment]
he @lingpri looks good but will you fix the build error or shall I do?

File: pages/alerts.rst [hunk 1]
```diff
@@ -312,6 +314,13 @@ Configuration Color
 Custom Message
     The message that will be sent to Slack.  The data described above can be used in this template.
 
+Backlog Message Items (optional)
```
Comment (waab76) [inline pages/alerts.rst]: Name here should match whatever is used across the 3 pages in the other PR.

File: pages/alerts.rst [hunk 2]
```diff
@@ -312,6 +314,13 @@ Configuration Color
...
+    The custom message has a data point named `backlog <http://localhost:8000/pages/alerts.html?highlight=alerts#backlog>`__, which gives
+    the preview of the messages that led to a slack notification.
```
Comment (waab76) [inline pages/alerts.rst]: I don't think we need to describe here what the message backlog is.  That should be discussed elsewhere in the Alert documentation.

File: pages/alerts.rst [hunk 3]
```diff
@@ -312,6 +314,13 @@ Configuration Color
...
+    Backlog Message Items will be used to override the Event definition Notification settings field `Message Backlog`
```
Comment (waab76) [inline pages/alerts.rst]: Again, name should match what is used on the UI (and I think should include the word `Limit` to make the purpose clear).

File: pages/alerts.rst [hunk 4]
```diff
@@ -312,6 +314,13 @@ Configuration Color
...
+    to limit the number of backlog messages sent as part of the slack message.
+    A Slack notification cannot have more than 50 backlog messages.
```
Comment (waab76) [inline pages/alerts.rst]: Open discussion on the other PR, but I don't think it makes any sense to enforce a max on the limit value.

**Review by waab76 — CHANGES_REQUESTED** [review]

**Comment by lingpri** [comment]
> he @lingpri looks good but will you fix the build error or shall I do?

@jalogisch - I have added a spacing, hopefully, its fixes the build issue.

File: pages/alerts.rst [hunk 5]
```diff
@@ -135,7 +135,10 @@ Event Data
         - ``priority`` (long) - The event's priority value.
         - ``fields`` (Map<String, String>) - The custom fields attached to the event.
 
+.. _Backlog:
+
 Backlog
+-------
```
Comment (waab76) [inline pages/alerts.rst]: Now that you're not linking to this from elsewhere in the documentation, I think you can get rid of the anchor and section heading.

```suggestion
Backlog
```

File: pages/alerts.rst [hunk 6]
```diff
@@ -312,6 +315,9 @@ Configuration Color
 Custom Message
     The message that will be sent to Slack.  The data described above can be used in this template.
 
+Message Backlog Limit (optional)
+    Limit the number of backlog messages sent as part of the Slack notification.
```
Comment (waab76) [inline pages/alerts.rst]: Should add something here to indicate that if the value is set to 0, no limit will be enforced (similar to the UI)

**Review by waab76 — CHANGES_REQUESTED** [review]

File: pages/alerts.rst (hunk 6 above)
Comment (lingpri) [reply pages/alerts.rst]: I have added the line in sync with UI help block. ` Limit the number of backlog messages sent as part of the Slack notification. If set to 0, no limit will be enforced.`

**Pull request merged**